cmd/livepeer/livepeer.go

```diff
@@ -506,6 +500,35 @@ func setupTranscoder(ctx context.Context, n *core.LivepeerNode, em eth.EventMoni
 		glog.Infof("Transcoder %v is active", n.Eth.Account().Address.Hex())
 	}
 
+	if publicAddr == "" {
+		// TODO probably should put this (along w wizard GETs) into common code
+		resp, err := http.Get("https://api.ipify.org?format=text")
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			glog.Error("Could not look up public IP address")
+			return err
+		}
+		publicAddr = strings.TrimSpace(string(body))
+	}
+	uriStr, err := n.Eth.GetServiceURI(n.Eth.Account().Address)
+	if err != nil {
+		glog.Error("Could not get service URI")
+		return err
+	}
+	uri, err := url.ParseRequestURI(uriStr)
+	if err != nil {
+		glog.Error("Could not parse service URI")
+		uri, _ = url.ParseRequestURI("http://127.0.0.1")
+	}
+	if uri.Hostname() != publicAddr {
+		glog.Errorf("Service address %v did not match discovered address %v; set the correct address in livepeer_cli or use -publicAddr", uri.Hostname(), publicAddr)
+		// TODO remove '&& false' after all transcoders have set a service URI
+		if active && false {
```

Comment:
Hm so after we remove the `&& false` to kill the node if an active transcoder has mismatched service URI + publicAddr if it is still the case that an active transcoder did not properly set a serviceURI to match their publicAddr they won't be able to use the CLI to set a service URI anymore since the transcoder configuration options would only be available when the node is running with the `-transcoder` flag, but if the account for the node is already an active transcoder without a service URI then every time the node boots up with the `-transcoder` flag the node will be killed. I guess the workaround here is that the user could pass in `-publicAddr 127.0.0.1` so that it matches the default service URI, successfully boot up the node, set an actual service URI and the reboot the node without the `-publicAddr` flag.

Not really a big concern IMO since the CLI makes you pick a service URI upon transcoder registration (the scenario would really only happen if someone registered as a transcoder outside of the CLI and chose to leave out the service URI), but just thought I would note the scenario.

Reply:
Correct, the `&&false` is to give currently active transcoders an opportunity to set up their service URIs; new transcoders registering after the network switchover wouldn't have this issue.

Also correct that `-publicAddr 127.0.0.1`  would be the fallback if current transcoders still don't set a service URI. Note that transcoders will still start up by default on port 4433 if there is no service URI set; see https://github.com/livepeer/go-livepeer/pull/447/files#diff-236217e7f473dfbde9134c6a391f3408R126

Since we have the fallback, we could still kill the node by default to force transcoders to make the adjustment, but I figured it's nicer to be a little less abrupt during the transition. We probably should monitor this though. The current transcoder pool is small enough that we could reach out to individual transcoders after a few weeks, if needed.